src/ui/src/builder/modals/UnsavedChangesModal.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+	<WdsModal
+		title="Unsaved Changes"
+		description="You have unsaved changes. What would you like to do?"
+		:actions="modalActions"
+		display-close-button
+		@close="() => emits('cancel')"
+	>
+		<div class="UnsavedChangesModal__content">
+			<div class="UnsavedChangesModal__icon">
+				<WdsIcon name="alert-triangle" />
+			</div>
+			<div class="UnsavedChangesModal__message">
+				<p>
+					The file <strong>{{ filename }}</strong> has unsaved
+					changes.
+				</p>
+				<p>Your changes will be lost if you don't save them.</p>
+			</div>
+		</div>
+	</WdsModal>
+</template>
+
+<script setup lang="ts">
+import { computed, withDefaults } from "vue";
+import WdsModal, { type ModalAction } from "@/wds/WdsModal.vue";
+import WdsIcon from "@/wds/WdsIcon.vue";
+
+interface UnsavedChangesModalProps {
+	filename: string;
+	isSaving?: boolean;
+}
+
+interface UnsavedChangesModalEmits {
+	save: [];
+	cancel: [];
+}
+
+const props = withDefaults(defineProps<UnsavedChangesModalProps>(), {
+	isSaving: false,
+});
+
+const emits = defineEmits<UnsavedChangesModalEmits>();
```

Comment:
We usually prefer object definition instead of TS. It's better because it does runtime checks.

```suggestion
const props = defineProps({
  filename: { type: String, required: true },
	isSaving: { type: boolean, required: false },
});

const emits = defineEmits({
  save: () => true,
  cancel: () => true,
});
```